src/test/java/net/ravendb/client/documents/queries/vectorSearch/VectorSearchTest.java

```diff
@@ -0,0 +1,1036 @@
+            throw new RuntimeException(e);
+        }
+    }
+
+    @EnableOn70Server
+    @Test
+    public void shouldGenerateRqlForBasicVectorSearchWithNumericEmbeddingValues() {
+        Float[] arr = new Float[]{2.5f, 3.3f};
+
+        try (IDocumentStore store = getDocumentStore()) {
+            try (IDocumentSession session = store.openSession()) {
+
+                String query = session.query(User.class)
+                        .vectorSearch(
+                                x-> x.withField("VectorField"),
+                                vf -> vf.byEmbedding(arr)
+                        )
+                        .toString();
+
+                assertEquals(
+                        "from 'Users' where vector.search(VectorField, $p0)",
+                        query
+                );
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @EnableOn70Server
+    @Test
+    public void shouldGenerateRqlForVectorSearchWithBase64EncodedEmbedding() {
+        try (IDocumentStore store = getDocumentStore()) {
+            try (IDocumentSession session = store.openSession()) {
+
+                String query = session.query(User.class)
+                        .vectorSearch(
+                                x-> x.withField("VectorField"),
+                                vf -> vf.byBase64("aaaa==")
+                        )
+                        .toString();
+
+                assertEquals(
+                        "from 'Users' where vector.search(VectorField, $p0)",
+                        query
+                );
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @EnableOn70Server
+    @Test
+    public void shouldGenerateRqlForVectorSearchWithTextFieldAndInt8Quantization() {
+        try (IDocumentStore store = getDocumentStore();
+             IDocumentSession session = store.openSession()) {
+
+            String query = session.query(User.class)
+                    .vectorSearch(
+                            x-> x.withText("EmbeddingSingles").targetQuantization(VectorEmbeddingType.INT8),
+                            vf -> vf.byText("aaaa")
+                    )
+                    .toString();
+
+            assertEquals(
+                    "from 'Users' where vector.search(embedding.text_i8(EmbeddingSingles), $p0)",
+                    query
+            );
+
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @EnableOn70Server
+    @Test
+    public void shouldGenerateRqlForVectorSearchUsingPropertySelectorForEmbeddingField() {
+        try (IDocumentStore store = getDocumentStore();
+             IDocumentSession session = store.openSession()) {
+
+            String query = session.query(User.class)
+                    .vectorSearch(
+                            x-> x.withEmbedding("EmbeddingSingles", null),
```

Comment:
why this null is needed?
```suggestion
                            x-> x.withEmbedding("EmbeddingSingles"),
```


https://github.com/M4xymm/ravendb-nodejs-client/blob/6abaeb6a1657a6f569513188636f46141f655e56/test/Documents/Queries/VectorSearchTest.ts#L85

Reply:
done introduced an overload